src/plugins/discover/public/application/main/hooks/use_esql_mode.ts

```diff
@@ -57,55 +58,61 @@ export function useEsqlMode({
           if (!query || next.fetchStatus === FetchStatus.ERROR) {
             return;
           }
+
           const sendComplete = () => {
             stateContainer.dataState.data$.documents$.next({
               ...next,
               fetchStatus: FetchStatus.COMPLETE,
             });
           };
+
           const { viewMode } = stateContainer.appState.getState();
-          let nextColumns: string[] = [];
           const isEsqlQuery = isOfAggregateQueryType(query);
-          const hasResults = Boolean(next.result?.length);
-          let queryHasTransformationalCommands = false;
-          if ('esql' in query) {
-            if (hasTransformationalCommand(query.esql)) {
-              queryHasTransformationalCommands = true;
-            }
-          }
 
           if (isEsqlQuery) {
-            const language = getAggregateQueryMode(query);
+            const hasResults = Boolean(next.result?.length);
+
             if (next.fetchStatus !== FetchStatus.PARTIAL) {
               return;
             }
 
+            let nextColumns: string[] = prev.current.recentlyUpdatedToColumns;
+
             if (hasResults) {
-              // check if state needs to contain column transformation due to a different columns in the resultset
+              // check if state needs to contain column transformation due to a different columns in the result set
```

Comment:
What I wrote here is no longer valid, and it wasn't good wording back in the days, maybe is better:
```suggestion
              // check if state needs to reset columns or set columns due to a transformational command
```
or just remove it

Reply:
removed